pkg/cloudprovider/types.go

```diff
@@ -31,8 +31,12 @@ type CloudProvider interface {
 	// GetInstanceTypes returns the instance types supported by the cloud
 	// provider limited by the provided constraints and daemons.
 	GetInstanceTypes(context.Context) ([]InstanceType, error)
+	// ValidateSpec is a hook for additional spec validation logic specific to the cloud provider.
+	// Note, implementations should not validate constraints resp. call `ValidateConstraints`
+	// from whithin this method as constraints are validated separately.
+	ValidateSpec(context.Context, *v1alpha2.ProvisionerSpec) *apis.FieldError
 	// Validate is a hook for additional constraint validation logic specific to the cloud provider
-	Validate(context.Context, *v1alpha2.Constraints) *apis.FieldError
+	ValidateConstraints(context.Context, *v1alpha2.Constraints) *apis.FieldError
```

Comment:
Can you add a note to the comment here about why we need this in addition to ValidateSpec, since constraints is a subfield. Something to the effect of,

"ValidateConstraints is used at provisioning time to ensure that pods are provisionable for the specified provisioner".